megatron/model/transformer.py

```diff
@@ -723,24 +792,33 @@ def forward(self, x, attention_mask, layer_past=None):
                 attention_output, presents = attention_output
                 self.layer_past = presents
             with torch.enable_grad():
-                attention_output = bias_dropout_fn(
-                    attention_output,
-                    bias=attention_bias.expand_as(residual),
-                    residual=residual,
-                    prob=self.hidden_dropout,
-                )
+                if attention_bias is not None:
+                    attention_output = bias_dropout_fn(
+                        attention_output,
+                        bias=attention_bias.expand_as(residual),
+                        residual=residual,
+                        prob=self.hidden_dropout,
+                    )
+                else:
+                    attention_output = attention_output + residual
 
             # output = x + mlp(ln2(x))
             mlp_output, mlp_bias = self.mlp(
                 self.post_attention_layernorm(attention_output)
             )
+
             with torch.enable_grad():
-                output = bias_dropout_fn(
-                    mlp_output,
-                    bias=mlp_bias.expand_as(attention_output),
-                    residual=attention_output,
-                    prob=self.hidden_dropout,
-                )
+                if self.mlp_type == "llama":
```

Comment:
I would rather call this "no bias MLPs" than "llama MLPs." That will make it much easier to follow this code for anyone who hasn't read the Llama paper (recently).

Reply:
In addition to not having bias terms, the architecture is also different (rather then up-proj then down-proj, there's an additional gating linear map). I don't feel strongly about the naming either way

Reply:
With the different architecture, I now like keeping the LLaMA-specific name.